### PR TITLE
setup Travis CI for testing

### PR DESCRIPTION
**Pull request by nathany**
- [x] It's still necessary to [activate a webhook](http://docs.travis-ci.com/user/getting-started/) to test pull requests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+go:
+  - 1.1
+  - 1.4
+
+before_script:
+  - FIXED=$(go fmt ./... | wc -l); if [ $FIXED -gt 0 ]; then echo "gofmt - $FIXED file(s) not formatted correctly, please run gofmt to fix this." && exit 1; fi
+
+script:
+  - go test -v ./...
```
Comment (nathany) [inline .travis.yml]: strangely that gives an error
```
can't load package: package gopkg.in/amz.v1: no buildable Go source files in /Users/nathany/src/gopkg.in/amz.v1
```

whereas `go test -v ./...` works fine.

**Comment by nathany** [comment]
It may be installing check.v1 with `-t` to the initial go get, but lets find out.

**Comment by nathany** [comment]
You can view the build output for this pull request here: https://travis-ci.org/go-amz/amz/jobs/45584945

The tests fail with:
```
ec2/ec2.go:25:2: cannot find package "gopkg.in/amz.v1/aws" in any of...
```

I don't think doing a `go get gopkg.in/amz.v1/...` is the right answer. That would be mixing code from the release version with code from the pull request under test.

In my limited experience with gopkg.in, I think we need to:

* [ ] have the `master` branch use internal imports with `github.com/go-amz/amz`
* [x] have the `v1` branch continue to use internal imports with `gopkg.in/amz.v1`
* [ ] have the `v2-dev` branch use `gopkg.in/amz.v2-dev` (once https://github.com/niemeyer/gopkg/issues/25 is done). currently v2-dev uses v1 code, also mixing code from different versions.
* [ ] Make `master` the default branch in GitHub, the default place pull requests are open to.
* [ ] Before tagging a release, merge over to the `v1` branch and ensure all import paths are correct.
* [ ] Advise users to import v1, but don't disallow use of or contribution to master (this is [what I do for fsnotify](https://github.com/go-fsnotify/fsnotify#api-stability), but my CONTRIBUTING doc could use some work).

If there is a better way to do this, I'd love to hear it (I have a tendency to overcomplicate things sometimes). /cc @niemeyer

File: .travis.yml (hunk 1 above)
Comment (dimitern) [inline .travis.yml]: Right, well - they both should do the same, but I forgot that's not actually true - I use -gocheck.v when running tests in a given subdir and it gives nicer output - it does choke when no *_test.go files exist in the root dir though.

Fair enough - no need to go get gocheck then.

**Comment by dimitern** [comment]
@nathany Nope, you're right - there is no top-level package or *.go files (go test -gocheck.v fails for the same reason). I should change the README to say:
```
go get gopkg.in/amz.v1/ec2
go get gopkg.in/amz.v1/aws
# alternatively ?:
go get gopkg.in/amz.v1/ec2/...
```
Then the travis script should go get each package the same way.

**Comment by nathany** [comment]
I think `go get gopkg.in/amz.v1/...` might actually do what you intend for the README. I'm starting to see those triple dots used in more places.

As for the Travis CI testing, if we can override the dest on Travis CI:
```
git clone --depth=50 git://github.com/go-amz/amz.git go-amz/amz
```

go gopkg.in/amz.v1

but still have it fetch the patch properly:

```
 git fetch origin +refs/pull/6/merge
```

Then we may not have to have different import statements on the master branch.

v2-dev might need a slightly different Travis CI file though.

Though the master branch is pretty much unusable in it's current state, it's really just there for https://github.com/niemeyer/gopkg/issues/20.

**Comment by nathany** [comment]
I sent an email to support@travis-ci.com

```
Hi,

I'm wondering if it is possible to override the git clone dest?

git clone --depth=50 git://github.com/go-amz/amz.git go-amz/amz

Instead I'd like to clone to:

gopkg.in/amz.v1

But still have additional fetches to test pull requests work.

If it's not possible, we have another option. I just want to out rule this before doing it.

The related issue on GitHub is:
https://github.com/go-amz/amz/pull/6

A reply there would be preferred.

Thanks,
Nathan.
```

**Comment by nathany** [comment]
Just FYI

> From December 23th to January 4th, the Travis CI team is taking a little bit of a break to spend time with their friends and families, so we may not be able to respond as timely as we usually do and support responses can be slower.

**Comment by nathany** [comment]
I took a look at https://github.com/go-mgo/mgo. It just has error.go on master so it doesn't use master at all. However there isn't any CI service setup for mgo, otherwise I'd borrow what's there. :smirk:

**Comment by dimitern** [comment]
@nathany Thanks for the update! This is very new to me as well, but I'm sure we'll sort out travis ci integration per branch - please share the response from Travis CI support when they reply.

So on go-amz/amz the default branch is v1, not master and before I deleted it by mistake it had a single error.go file like mgo has. However, go get gopkg.in/amz (IIRC this should be equivalent to amz.v0 - i.e. master) still got me v1. Also running git clone https://github.com/go-amz/amz.git clones branch v1 by default.

**Comment by nathany** [comment]
> Hey Nathan,

> Thanks for your email.

> The clone destination currently can't be changed, I'm afraid.

> Cheers, Mathias

> --
> Mathias Meyer
> support@travis-ci.com

**Comment by nathany** [comment]
Hm.

Maybe a well placed command could move the TRAVIS_BUILD_DIR before running the build.

**Comment by nathany** [comment]
@dimitern I got it working and rebased all my attempts down to a nice single commit that looks like I know what I'm doing. :-D

PTAL

**Comment by nathany** [comment]
FYI, the tests failed on Go 1.1, Go 1.2+ is required:

```
aws/sign.go:278: undefined: sha256.Sum256
```

https://travis-ci.org/go-amz/amz/jobs/45743738

**Comment by dimitern** [comment]
@nathany Sweet! Thanks for making it work :)
I don't think @niemeyer would mind merging this, as it's not related to code changes, so I'll do it now.
Then, I'll fix the master branch.

**Comment by nathany** [comment]
Thanks @dimitern.

The beauty of source control, it's easy to revert later if necessary.
But hopefully this works well.

It will need some tweaking for the v2-dev branch.

**Pull request merged**